kcp/deploy.sh

```diff
@@ -0,0 +1,295 @@
+          echo "Unsupported environment: $1" >&2
+          echo "" >&2
+          usage
+          exit 1
+        fi
+        ;;
+      esac
+      ;;
+    -k | --kcp)
+      shift
+      case "$1" in
+      stable | unstable)
+        KCP_ENV="kcp-$1"
+        ;;
+      *)
+        echo "Unsupported environment: $1" >&2
+        echo "" >&2
+        usage
+        exit 1
+        ;;
+      esac
+      ;;
+    -w | --workspace)
+      shift
+      KCP_WORKSPACE="$1"
+      ;;
+    -d | --debug)
+      set -x
+      DEBUG="-d"
+      ;;
+    -h | --help)
+      usage
+      exit 0
+      ;;
+    --)
+      # End of arguments
+      break
+      ;;
+    *)
+      echo "Unknown argument: $1"
+      usage
+      exit 1
+      ;;
+    esac
+    shift
+  done
+}
+
+argocd_local() {
+    mkdir -p "$ARGOCD_DIR"
+    argocd --config "$ARGOCD_DIR/config" "$@"
+}
+
+kcp_config() {
+  mkdir -p "$KUBECONFIG_DIR"
+  KUBECONFIG="$KUBECONFIG_KCP" "$@"
+}
+
+plnsvc_config() {
+  mkdir -p "$KUBECONFIG_DIR"
+  KUBECONFIG="$KUBECONFIG_PLNSVC" "$@"
+}
+
+configure_pipeline_cluster() {
+  echo "[Pipeline cluster]"
+
+  echo "  - Setup the cluster:"
+  plnsvc_config kubectl apply -f "$SCRIPT_DIR/manifests/plnsvc/namespace.yaml"
+
+  echo "  - Service account for connecting KCP to the plnsvc cluster:"
+  plnsvc_config kubectl apply -f "$SCRIPT_DIR/manifests/plnsvc/kcp-manager.yaml" --wait
+  get_context plnsvc_config kcp pln-svc kcp-manager "$KUBECONFIG_PLNSVC_KCP"
+
+  echo
+}
+
+configure_kcp_cluster() {
+  echo "[KCP cluster]"
+
+  echo "  - Service account for connecting the pipelines controller to KCP: "
+  kcp_config kubectl apply -f "$SCRIPT_DIR/manifests/kcp/pln-svc-manager.yaml" --wait
+  get_context kcp_config "$KCP_ENV-pln-svc" pln-svc pln-svc-manager "$KUBECONFIG_KCP_PLNSVC"
+
+  # Setup the resources required for ArgoCD to manage the KCP cluster
+  # Check if this can be deprecated when ArgoCD runs on KCP instead of the Pipeline cluster
+  echo "  - Service account for connecting KCP to ArgoCD: "
+  kcp_config kubectl apply -f "$SCRIPT_DIR/manifests/kcp/argocd-manager.yaml" --wait
+  get_context kcp_config "$KCP_ENV-argocd" kube-system argocd-manager "$KUBECONFIG_KCP_ARGOCD"
+
+  echo -n "  - Create plnsvc workloadcluster: "
+  local manifests_dir="$WORK_DIR/manifests"
+  mkdir -p "$manifests_dir/plnsvc"
+  if ! kcp_config kubectl get workloadcluster plnsvc >/dev/null 2>&1; then
+    kcp_config kubectl kcp workload sync plnsvc --kcp-namespace pln-svc \
+      --resources pods,services \
+      --syncer-image ghcr.io/kcp-dev/kcp/syncer-c2e3073d5026a8f7f2c47a50c16bdbec:41ca72b > "$manifests_dir/plnsvc/kcp-syncer.yaml"
+  fi
+  echo "OK"
+
+  echo
+}
+
+
+register_clusters() {
+  echo "[Clusters registration]"
+
+  echo "  - Register the plnsvc cluster to KCP: "
+  if [ -f "$WORK_DIR/manifests/plnsvc/kcp-syncer.yaml" ]; then
+    plnsvc_config kubectl apply -f "$WORK_DIR/manifests/plnsvc/kcp-syncer.yaml" --wait
```

Comment:
--wait does nothing for 'kubectl apply'. Please remove it in all the places.

Reply:
Fixed